src/test/java/io/axoniq/axonserver/connector/event/impl/BufferedAggregateEventStreamTest.java

```diff
@@ -66,4 +66,37 @@ void testEventStreamPropagatesErrorOnHasNextAfterReadingAvailableEvents() throws
         assertThrows(StreamClosedException.class, () -> testSubject.hasNext());
     }
 
-}
\ No newline at end of file
+    @Test
+    void throwsExceptionOnTimeoutWhileRetrievingEvents() throws Exception {
+        reduceTimeout();
+
+        // Push messages
+        for (int i = 0; i < 20; i++) {
+            testSubject.onNext(Event.newBuilder().setAggregateSequenceNumber(i).build());
+            testSubject.hasNext();
+            testSubject.next();
+        }
+
+        // Now, wait while there is no message
```

Comment:
You're not necessarily waiting here, right? The for-loop already consumes the entries through the `next` operation, right? Ergo, this `hasNext` call is already confirmed to be at the end of the stream.

Reply:
Yes, we are waiting in the `hasNext` call, expecting a throw there due to a timeout. It is waiting because all the entries have been consumed and are the stream is not closed (indicated by a terminal message in the buffer)